mne/io/base.py

```diff
@@ -1933,6 +1939,10 @@ def _write_raw(fname, raw, info, picks, fmt, data_type, reset_range, start,
     if first_samp != 0:
         write_int(fid, FIFF.FIFF_FIRST_SAMPLE, first_samp)
 
+    if raw.annotations is not None:
+        annotations = raw.annotations._serialize()
+        write_string(fid, FIFF.FIFF_MNE_ANNOTATIONS, annotations)
```

Comment:
Then instead of just dumping a string, you would make `FIFF_MNE_ANNOTATIONS` a block type instead, and write out the tags in the block the standard way.